src/test/java/org/jenkinsci/plugins/durabletask/PowershellScriptTest.java

```diff
@@ -65,7 +66,7 @@
         String pathSeparator = properties.getProperty("path.separator");
         String[] paths = System.getenv("PATH").split(pathSeparator);
         boolean powershellExists = false;
-        String cmd = launcher.isUnix()?"powershell":"powershell.exe";
+        String cmd = launcher.isUnix()?"pwsh":"powershell.exe";
```

Comment:
Ahh and here be dragons...
People are using cygwin sshd to launch slaves on windows which I think ends up telling Jenkins that the OS is Unix, and well the powershell executable is `powershell` and not `pwsh` that it is on linux....

given that this will break any existing setup I am probably against this change here.

Reply:
Good point. I did not consider Cygwin here. How would you prefer this to be handled? I could test for both binaries regardless of platform, but then the issue is that I assume pwsh for *nix and powershell.exe for windows in \src\main\java\org\jenkinsci\plugins\durabletask\PowershellScript.java, so I'd need extra logic to determine which binary should be used, which seems like extra overhead for just running a script.

Is there a way to detect if Cygwin is being used?